### PR TITLE
Sync v2-6-stable with v2-6-test to release 2.6.2

### PR DESCRIPTION
**Pull request by eladkal**
Time for `2.6.2rc1`!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,94 @@
...
+- Fix version support information (#31468)
+- Add missing BashOperator import to documentation example (#31436)
+- Update the output of airflow info command in the doc (#31336)
+- Fix task.branch error caused by incorrect initial parameter (#31265)
+- Update callbacks documentation (errors and context) (#31116)
+- Add an example for dynamic task mapping with non-TaskFlow operator (#29762)
+- Few doc fixes - links, grammer and wording (#31719)
+- Add description in a few more places about adding airflow to pip install (#31448)
+
+Improvement Changes
```
Comment (pierrejeambrun) [inline RELEASE_NOTES.rst]: I think semver requires a minor release for this, we can either mark them for 2.7.0 and drop them from the cherry picking, or consider them as 'misc/internal' if we want them in this patch. (if applicable)

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,94 @@
...
+- Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments. (#31308)
+- Use only one line for tmp dir log (#31170)
+- Executor events are not always "exited" here (#30859)
+- Rephrase comment in setup.py (#31312)
+
+Doc-only changes
+^^^^^^^^^^^^^^^^
+- update path of KubernetesPodOperator in core docs (#31683)
+- Fix wrong cron syntax (#31666)
+- 📝 Update release documentation (#31631)
```
Comment (pierrejeambrun) [inline RELEASE_NOTES.rst]: Not sure how this icon will render on our website, mails and stuff. I would be for removing it

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,94 @@
...
+- Skip auto-document things with :meta private: (#31380)
+- Add an example for sql_alchemy_connect_args conf (#31332)
+- Convert dask upper-binding into exclusion (#31329)
+- Upgrade FAB to 4.3.1 (#31203)
+- Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments. (#31308)
+- Use only one line for tmp dir log (#31170)
+- Executor events are not always "exited" here (#30859)
+- Rephrase comment in setup.py (#31312)
+
+Doc-only changes
```
Comment (pierrejeambrun) [inline RELEASE_NOTES.rst]:
```suggestion
Doc only changes
```

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (eladkal) [reply RELEASE_NOTES.rst]: All can be classified as misc I just thought they are a bit more important to note for the users

Comment (pierrejeambrun) [reply RELEASE_NOTES.rst]: Yes, lets see what others think

Comment (eladkal) [reply RELEASE_NOTES.rst]: if the issue is just about the title we can change it.. if the issue is about a specific PR that shouldn't be included in this release that is another issue :)

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,6 +21,94 @@
...
+
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Fix Kubernetes executor set wrong task status (#31274)
+- Use keyword parameters for migration methods for mssql (#31309)
+- Control permissibility of driver config in extra from airflow.cfg (#31754)
+- Retain on_failure_fail_dagrun attr when other task attrs are overridden (#31678)
+- fixing broken links in openapi/v1.yaml (#31619)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Fixing broken links in openapi/v1.yaml (#31619)
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,6 +21,94 @@
...
+- Hide old alert box when testing conn with different value (#31606)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Hide old alert box when testing connection with different value (#31606)
```

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,6 +21,94 @@
...
+- Add TriggererStatus to OpenAPI spec (#31579)
+- Resolving issue where Grid won't un-collapse when Details is collapsed (#31561)
+- Fix sorting of tags (#31553)
+- Add the missing ``map_index`` to the xcom key when skipping downstream tasks (#31541)
+- fix airflow users delete CLI command (#31539)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix airflow users delete CLI command (#31539)
```

File: RELEASE_NOTES.rst [hunk 7]
```diff
@@ -21,6 +21,94 @@
...
+- Include triggerer health status in Airflow /health endpoint (#31529)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Include triggerer health status in Airflow `/health` endpoint (#31529)
```

File: RELEASE_NOTES.rst [hunk 8]
```diff
@@ -21,6 +21,94 @@
...
+- Rephrase comment in setup.py (#31312)
+
+Doc only changes
+^^^^^^^^^^^^^^^^
+- update path of KubernetesPodOperator in core docs (#31683)
+- Fix wrong cron syntax (#31666)
+- Update release documentation (#31631)
+- Add docstring and signature for _read_remote_logs (#31623)
+- Updating UI screenshots in docs (#31577)
+- Remove typo in docs serializers.rst (#31507)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
```
I think we should remove these, doesn't seem important

Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
```
I think we should remove these, doesn't seem important

File: RELEASE_NOTES.rst [hunk 9]
```diff
@@ -21,6 +21,94 @@
...
+- Remove note about triggerer being 3.7+ only (#31483)
+- Fix version support information (#31468)
+- Add missing BashOperator import to documentation example (#31436)
+- Update the output of airflow info command in the doc (#31336)
+- Fix task.branch error caused by incorrect initial parameter (#31265)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst:95]:
```suggestion
```
This is breeze related. We can skip

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst]: Most of the items in the improvement looks like real improvements and I suggest we move them to 2.7.0. I think we should be using patch releases only for bug fixes.

File: RELEASE_NOTES.rst (hunk 9 above)
Comment (eladkal) [reply RELEASE_NOTES.rst:95]: it's not breeze but we can remove it if you'd like..
it's a doc only change

File: RELEASE_NOTES.rst [hunk 10]
```diff
@@ -21,6 +21,90 @@
...
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+No significant changes.
+
+Bug Fixes
+^^^^^^^^^
+- Fix Kubernetes executor set wrong task status (#31274)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix Kubernetes executors detection of deleted pods (#31274)
```

File: RELEASE_NOTES.rst [hunk 11]
```diff
@@ -21,6 +21,90 @@
...
+- Use keyword parameters for migration methods for mssql (#31309)
+- Control permissibility of driver config in extra from airflow.cfg (#31754)
+- Retain ``on_failure_fail_dagrun`` attr when other task attrs are overridden (#31678)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
```

This should wait for 2.7, it's part of AIP-52.

File: RELEASE_NOTES.rst [hunk 12]
```diff
@@ -21,6 +21,90 @@
...
+- Remove found_descendents param from get_flat_relative_ids (#31559)
+- Fix typing in external task triggers (#31490)
+- Wording the next and last run DAG columns better (#31467)
+- Skip auto-document things with :meta private: (#31380)
+- Add an example for sql_alchemy_connect_args conf (#31332)
+- Convert dask upper-binding into exclusion (#31329)
+- Upgrade FAB to 4.3.1 (#31203)
+- Added metavar and choices to --state flag in airflow dags list-jobs CLI for suggesting valid state arguments. (#31308)
+- Use only one line for tmp dir log (#31170)
+- Executor events are not always "exited" here (#30859)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This feels like a bugfix to me.

File: RELEASE_NOTES.rst [hunk 13]
```diff
@@ -21,6 +21,90 @@
...
+- Rephrase comment in setup.py (#31312)
+- Add fullname to owner on logging (#30185)
+- Validate connection IDs (#31140)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This also feels like a bugfix.

File: RELEASE_NOTES.rst [hunk 14]
```diff
@@ -21,6 +21,90 @@
...
+- Make connection id validation consistent across interface (#31282)
+- Support for sorting DAGs by Last Run Date in the web UI (#31234)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: Was this brought in for cherrypicking ease? This feels out of place in a patch release.

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (jedcunningham) [reply RELEASE_NOTES.rst]: I could see reclassifying the 3.11 one as misc, but I think the other 2 should wait.

File: RELEASE_NOTES.rst (hunk 14 above)
Comment (eladkal) [reply RELEASE_NOTES.rst]: Its a 2 line change. We can push it to 2.7.0
(In some casses we have a thin line between a misc change and a feature for example adding a templated_field)

File: RELEASE_NOTES.rst (hunk 9 above)
Comment (ephraimbuddy) [reply RELEASE_NOTES.rst:95]: Sorry, it was the one below it that I was referring to, it has more of breeze change. Let's remove it

**Comment by potiuk** [comment]
FYI @eladkal  The CI failures with Python 3.11 are likely because you need to cherry-pick: https://github.com/apache/airflow/pull/31813. While it does not impact the production image build, it makes a difference when the CI image is created because `[devel-ci]` extra is calculated using the dependencies of all providers that are part of the sources.

**Comment by pierrejeambrun** [comment]
Thanks Jarek, I was investigating that.

> FYI @eladkal The CI failures with Python 3.11 are likely because you need to cherry-pick: https://github.com/apache/airflow/pull/31813. While it does not impact the production image build, it makes a difference when the CI image is created because [devel-ci] extra is calculated using the dependencies of all providers that are part of the sources

The PR linked seems wrong :thinking:

**Comment by potiuk** [comment]
ah yeah :). Wrong copy&paste https://github.com/apache/airflow/pull/30067

File: RELEASE_NOTES.rst (hunk 11 above)
Comment (eladkal) [reply RELEASE_NOTES.rst]: removed

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (eladkal) [reply RELEASE_NOTES.rst]: I removed/reclassified all the PRs that were under improvements.

File: RELEASE_NOTES.rst (hunk 14 above)
Comment (eladkal) [reply RELEASE_NOTES.rst]: pushed this one to 2.7.0

**Review by potiuk — APPROVED** [review]
LGTM

**Comment by potiuk** [comment]
Shoudl we also bump 2.6.1 -> 2.6.2 in `pre_commit_supported_versions` (and as a consequence in other places it updates?)

**Comment by pierrejeambrun** [comment]
> Should we also bump 2.6.1 -> 2.6.2 in pre_commit_supported_versions.py (and as a consequence in other places it updates?)

Yes, Elad did that in 47438559e3, but was overriden when syncing dev scripts with main in `9323ab1e2b`.

I'll bring that at the tip of the branch and make sure we have everything needed for bumping to 2.6.2

**Comment by potiuk** [comment]
> Yes, Elad did that in https://github.com/apache/airflow/commit/47438559e349a4c6d0ebf55d7ff548cd1ddca96b, but was overriden when syncing dev scripts with main in https://github.com/apache/airflow/commit/9323ab1e2b2570ad98b2bea464a94639cc8e1748.

Thought so :)

**Comment by pierrejeambrun** [comment]
Should be good

**Review by pierrejeambrun — APPROVED** [review]

**Review by ephraimbuddy — APPROVED** [review]

**Comment by potiuk** [comment]
We also need to cherry-pick https://github.com/apache/airflow/pull/31839 since the next build will break breeze unit tests due to `pytest` breaking change in patch-level 7.3.2 release yesterday. I will do it

**Pull request merged**